### ISSUE TITLE
[Bug]: Screensaver should not be inhibited when paused

### ISSUE THREAD

**Issue by kc2zgu** [OpenTTD/OpenTTD#12937]
### Version of OpenTTD

13.4 Linux

### Expected result

When running the game, the screensaver is inhibited so that it does not unexpectedly interrupt play (it looks like this is done in side SDL by default, see https://wiki.libsdl.org/SDL2/SDL_DisableScreenSaver, since in a mouse/keyboard-based game there is little need to explicitly do this and OpenTTD gets it "for free"). However, there is no reason to keep the screensaver disabled when the game is paused. It effectively just wastes energy, especially for games played over multiple days when the program is left running overnight but paused.

### Actual result

Screensaver should be re-enabled when the game is paused and disabled again after unpausing, or not disabled at all if the feature is not necessary.

### Steps to reproduce

1. Start OpenTTD
2. Pause the game
3. Wait for screensaver to activate

**Comment by TrueBrain** [comment]
Just as note: one playstyle OpenTTD offers is "build when paused", where people pause the game to build complete networks. So it isn't as easy as saying: enable screensaver when paused.

**Comment by PeterN** [comment]
I would say we should tell SDL not to disable the screensaver at all. That is only needed for joystick controller input games.

**Comment by TrueBrain** [comment]
Inwould absolutely hate it when the screensaver activates when I am just watching the game for a bit.

In fact, most games I play don't allow screensaver to enable, even when paused. Seems like a sane default to have?

**Comment by krysclarke** [comment]
Perhaps the solution is to introduce a setting to toggle inhibiting the screensaver (defaulting to the current choice of inhibiting the screensaver)?
I'm not sure if OpenRCT2 uses SDL but it does have such a setting in the GUI, so I'm guessing it's possible for it to be implemented in OpenTTD, too.

**Comment by kc2zgu** [comment]
> I would say we should tell SDL not to disable the screensaver at all. That is only needed for joystick controller input games.

Right, since OpenTTD uses mouse/keyboard input the screensaver won't activate while someone is actively playing. Even when using build while paused, there will still be mouse and keyboard input. Watching the game run for an extended period of time seems like a niche use case to me, but disabling screensaver when not paused would allow this to work. A much more common use case, at least for me, involves leaving the program running for sometimes long periods of time (e.g. overnight) with the game paused; there's no good reason to keep the screen on and prevent power saving in this case.

Or it could be a setting people can change so all these play styles could be made to work.

**Comment by 2TallTyler** [comment]
I am closing this ticket as there's not a clear bug here: the current behavior is intentional. Feel free to open a PR to add a setting (or solve in another way), but this isn't a bug. 🙂
